File: praca-inzynierska-frontend/src/app/app/model/PreferenceType.ts
export enum PreferenceType {
    WILLING = 'WILLING',
    AVAILABLE = 'AVAILABLE',
    UNWILLING = 'UNWILLING',
    UNAVAILABLE = 'UNAVAILABLE'
}

export namespace PreferenceType {

    export const DEFAULT = PreferenceType.AVAILABLE

    export function toColor(type: PreferenceType): string { 
        switch(type) {
            case PreferenceType.WILLING:
                return '#89c97b'
            case PreferenceType.AVAILABLE:
                return '#cbfcc0'
            case PreferenceType.UNWILLING:
                return '#ebd071'
            case PreferenceType.UNAVAILABLE:
                return '#fa6e6e'
            default:
                return '#ffffff'
        }
    }

}

File: praca-inzynierska-backend/schedule-generator/src/main/kotlin/jwer/schedulegenerator/generator/model/PreferenceType.kt
package jwer.schedulegenerator.generator.model

enum class PreferenceType {
    UNAVAILABLE,
    UNWILLING,
    AVAILABLE,
    WILLING;

    companion object {

        fun fromNumber(number: Int): PreferenceType? {
            return when (number) {
                0 -> UNAVAILABLE
                1 -> UNWILLING
                2 -> AVAILABLE
                3 -> WILLING
                else -> null
            }
        }

    }
}